Fix WW3/Global Workflow debug build warnings and postprocess errors

mingchen-NOAA opened the pull request:
# Pull Request Summary
<!-- A short overview of the PR --> 
This PR resolved WW3 build warnings and postprocess errors for debug mode of global workflow.
## Description
<!--
Provide a detailed description of what this PR does.
What bug does it fix, or what feature does it add?
Is a change of answers expected from this PR?

Please also include the following information: 
* Add any suggestions for a reviewer 
* Mention any labels that should be added:  _bug_, _documentation_, _enhancement_, _new feature_
* Are answer changes expected from this PR? Please describe the changes and the reason why in addition to which of the following labels would apply: _mod_def change_, _out_grd change_, _out_pnt change_, _restart file change_, _Regression test_ 
-->
This work addresses Intel compiler warning #8889 messages when building ww3_prepost and ufs_gfs components of the global workflow in debug mode. In addition, it fixes runtime errors encountered during wave postprocessing after a global workflow forecast run.

**_Ensure the production GFS WW3 code compiles without warnings_** (tied to the issue #1500):
- Explicit EXTERNAL declarations were added in the following source files: [w3profsmd.F90, w3strkmd.F90, ww3_bounc.F90, ww3_grib.F90, ww3_prnc.F90, ww3_systrk.F90,](https://github.com/NOAA-EMC/WW3/pull/1504/commits/e495cda03a348ccd63c50364b2f5ca78941d70fa) and [yowfunction.F90, yowpdlibmain.F90](https://github.com/NOAA-EMC/WW3/pull/1504/commits/6750110add2636ecdc2260bee9333ec1e757fc4b). The update resolve Intel compiler warning #8889.

**_Ensure that all WW3-related jobs can be ran in the debug mode without errors_** (tied to the issue #1502):
- Character variables MAPICET, MAPDRYT, MAPLNDT, and MAPMSKT were initialized in source file ww3_gint.F90 and WORDS was initialized in ww3_grib.F90 to resolve the run-time check failures for the [wavepostsbs job](https://github.com/NOAA-EMC/WW3/pull/1504/commits/c2d914d35542064ba9f7dc6905c9c46c2b4b2d32). 
- Write WORDS statement was also removed to fix the ^@ artifacts in the log files. 
- Formatted writes in w3arrymd.F90 and ww3_outp.F90 were adjusted to prevent output conversion errors.

Please also include the following information: 
* Add any suggestions for a reviewer:
* Mention any labels that should be added: _enhancement and bug_
* Are answer changes expected from this PR? No scientific answer changes are expected. Only output logs (ww3_*.out) will change

### Issue(s) addressed
<!--
* Please list any issues associated with this PR, including those the PR will fix/close. For example:  
- fixes #<issue_number>
- fixes noaa-emc/ww3/issues/<issue_number>
-->
- fixes #1500 
- fixes #1502 
### Commit Message
<!--
Please provide a short summary of this PR, which will be used during _Squash and Merge_ and will be shown as a git log message.  Be sure to add any co-authors here. 
-->
Fix WW3/Global Workflow debug build warnings and postprocess errors
### Check list  

<!-- After creating the PR you can check each of the items below that have been completed -->

- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [x] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested? matrix
* Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) yes
* Have the matrix regression tests been run (if yes, please note HPC and compiler)? Hercules Intel
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.) log files as ww3_.out 
* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):

**Expected differences from known non-identical tests** (note: these differences also include ww3_*.out changes due to this PR):
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR1_MPI_e                     (2 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (2 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (2 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (17 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e_c                     (2 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (2 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (18 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (21 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (18 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (2 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (2 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (18 files differ)
mww3_test_09/./work_MPI_ASCII                     (3 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (8 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.6/./work_ST4_ASCII                     (1 files differ)
ww3_ufs1.3/./work_a                     (8 files differ)
```

**Expected differences due to this PR:**
All changes are log files (ww3_*.out). No scientific answer changes in binary or netCDF files.
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_01/./work_PR2_UQ                     (4 files differ)
mww3_test_01/./work_PR3_UNO                     (4 files differ)
mww3_test_01/./work_PR2_UQ_MPI                     (4 files differ)
mww3_test_01/./work_PR3_UNO_MPI                     (4 files differ)
mww3_test_01/./work_PR3_UQ                     (4 files differ)
mww3_test_01/./work_PR3_UQ_MPI                     (4 files differ)
mww3_test_01/./work_PR2_UNO_MPI                     (4 files differ)
mww3_test_01/./work_PR1                     (4 files differ)
mww3_test_01/./work_PR2_UNO                     (4 files differ)
mww3_test_01/./work_PR1_MPI                     (4 files differ)
mww3_test_02/./work_PR2_UQ_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_d                     (2 files differ)
mww3_test_02/./work_PR2_UQ_MPI_c                     (2 files differ)
mww3_test_02/./work_PR3_UNO_b                     (2 files differ)
mww3_test_02/./work_PR3_UQ_b_c                     (2 files differ)
mww3_test_02/./work_PR2_UNO_a                     (2 files differ)
mww3_test_02/./work_PR1_d                     (2 files differ)
mww3_test_02/./work_PR3_UQ_a_c                     (2 files differ)
mww3_test_02/./work_PR1_MPI_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_SCRIPMPI                     (2 files differ)
mww3_test_02/./work_PR2_UNO_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_a                     (2 files differ)
mww3_test_02/./work_PR2_UNO_MPI_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_c                     (2 files differ)
mww3_test_02/./work_PR1_MPI_c                     (2 files differ)
mww3_test_02/./work_PR2_UQ_a                     (2 files differ)
mww3_test_02/./work_PR2_UQ_d                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_d                     (2 files differ)
mww3_test_02/./work_PR3_UNO_d_c                     (2 files differ)
mww3_test_02/./work_PR2_UNO_MPI_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_a                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_a_c                     (2 files differ)
mww3_test_02/./work_PR1_a                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_d_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_b_c                     (2 files differ)
mww3_test_02/./work_PR2_UNO_MPI_a                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_a_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_d_c                     (2 files differ)
mww3_test_02/./work_PR2_UQ_MPI_d                     (2 files differ)
mww3_test_02/./work_PR3_UQ_b                     (2 files differ)
mww3_test_02/./work_PR3_UQ_d                     (2 files differ)
mww3_test_02/./work_PR3_UQ_c                     (2 files differ)
mww3_test_02/./work_PR2_UQ_MPI_b                     (2 files differ)
mww3_test_02/./work_PR2_UNO_MPI_d                     (2 files differ)
mww3_test_02/./work_PR1_MPI_a                     (2 files differ)
mww3_test_02/./work_PR1_b                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_d_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_MPI_d                     (2 files differ)
mww3_test_02/./work_PR1_c                     (2 files differ)
mww3_test_02/./work_PR2_UQ_c                     (2 files differ)
mww3_test_02/./work_PR2_UQ_MPI_a                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_a                     (2 files differ)
mww3_test_02/./work_PR3_UNO_a_c                     (2 files differ)
mww3_test_02/./work_PR3_UNO_c                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_b_c                     (2 files differ)
mww3_test_02/./work_PR3_UQ_a                     (2 files differ)
mww3_test_02/./work_PR2_UNO_b                     (2 files differ)
mww3_test_02/./work_PR3_UNO_b_c                     (2 files differ)
mww3_test_02/./work_PR3_UNO_MPI_c                     (2 files differ)
mww3_test_02/./work_PR1_MPI_d                     (2 files differ)
mww3_test_02/./work_PR2_UNO_d                     (2 files differ)
mww3_test_03/./work_PR3_UQ_c_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_b                     (3 files differ)
mww3_test_03/./work_PR3_UNO_d                     (3 files differ)
mww3_test_03/./work_PR2_UQ_MPI_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_b                     (3 files differ)
mww3_test_03/./work_PR3_UQ_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_b_c                     (3 files differ)
mww3_test_03/./work_PR2_UNO_a                     (2 files differ)
mww3_test_03/./work_PR1_e                     (1 files differ)
mww3_test_03/./work_PR1_d                     (3 files differ)
mww3_test_03/./work_PR3_UQ_a_c                     (2 files differ)
mww3_test_03/./work_PR1_MPI_b                     (3 files differ)
mww3_test_03/./work_PR2_UNO_c                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_a                     (2 files differ)
mww3_test_03/./work_PR2_UNO_MPI_c                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_c                     (3 files differ)
mww3_test_03/./work_PR1_MPI_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_a                     (2 files differ)
mww3_test_03/./work_PR2_UQ_d                     (3 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d                     (3 files differ)
mww3_test_03/./work_PR3_UNO_d_c                     (3 files differ)
mww3_test_03/./work_PR2_UNO_MPI_b                     (3 files differ)
mww3_test_03/./work_PR3_UNO_a                     (2 files differ)
mww3_test_03/./work_PR3_UQ_MPI_b                     (3 files differ)
mww3_test_03/./work_PR3_UNO_MPI_a_c                     (2 files differ)
mww3_test_03/./work_PR1_a                     (2 files differ)
mww3_test_03/./work_PR3_UNO_MPI_b                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_c_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_c_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d_c                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_b_c                     (3 files differ)
mww3_test_03/./work_PR2_UNO_MPI_a                     (2 files differ)
mww3_test_03/./work_PR2_UQ_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_a_c                     (2 files differ)
mww3_test_03/./work_PR3_UQ_d_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_c_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d                     (3 files differ)
mww3_test_03/./work_PR3_UQ_b                     (3 files differ)
mww3_test_03/./work_PR3_UQ_d                     (3 files differ)
mww3_test_03/./work_PR3_UQ_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_MPI_b                     (3 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d                     (3 files differ)
mww3_test_03/./work_PR1_MPI_a                     (2 files differ)
mww3_test_03/./work_PR1_b                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d_c                     (3 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d                     (3 files differ)
mww3_test_03/./work_PR1_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_c                     (3 files differ)
mww3_test_03/./work_PR2_UQ_MPI_a                     (2 files differ)
mww3_test_03/./work_PR3_UNO_MPI_a                     (2 files differ)
mww3_test_03/./work_PR3_UQ_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_a_c                     (2 files differ)
mww3_test_03/./work_PR3_UNO_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_MPI_b_c                     (3 files differ)
mww3_test_03/./work_PR3_UQ_a                     (2 files differ)
mww3_test_03/./work_PR2_UNO_b                     (3 files differ)
mww3_test_03/./work_PR3_UNO_b_c                     (3 files differ)
mww3_test_03/./work_PR3_UNO_MPI_c                     (3 files differ)
mww3_test_03/./work_PR1_MPI_d                     (3 files differ)
mww3_test_03/./work_PR2_UNO_e                     (1 files differ)
mww3_test_03/./work_PR3_UNO_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_d                     (3 files differ)
mww3_test_04/./work_PR2_UQ_b                     (1 files differ)
mww3_test_04/./work_PR3_UNO_MPI_e_c                     (1 files differ)
mww3_test_04/./work_PR2_UQ_MPI_c                     (1 files differ)
mww3_test_04/./work_PR3_UNO_b                     (1 files differ)
mww3_test_04/./work_PR3_UQ_g                     (1 files differ)
mww3_test_04/./work_PR3_UQ_b_c                     (1 files differ)
mww3_test_04/./work_PR2_UNO_a                     (2 files differ)
mww3_test_04/./work_PR1_f                     (1 files differ)
mww3_test_04/./work_PR3_UQ_a_c                     (2 files differ)
mww3_test_04/./work_PR1_MPI_b                     (1 files differ)
mww3_test_04/./work_PR2_UNO_g                     (1 files differ)
mww3_test_04/./work_PR2_UNO_MPI_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_c                     (1 files differ)
mww3_test_04/./work_PR1_MPI_c                     (1 files differ)
mww3_test_04/./work_PR2_UQ_a                     (2 files differ)
mww3_test_04/./work_PR3_UNO_MPI_d                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_04/./work_PR2_UNO_f                     (1 files differ)
mww3_test_04/./work_PR2_UNO_MPI_b                     (1 files differ)
mww3_test_04/./work_PR3_UNO_a                     (2 files differ)
mww3_test_04/./work_PR3_UQ_MPI_b                     (1 files differ)
mww3_test_04/./work_PR1_a                     (2 files differ)
mww3_test_04/./work_PR3_UNO_MPI_b                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_c_c                     (1 files differ)
mww3_test_04/./work_PR3_UNO_MPI_c_c                     (1 files differ)
mww3_test_04/./work_PR3_UNO_MPI_d_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_b_c                     (1 files differ)
mww3_test_04/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_04/./work_PR3_UQ_f_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_g_c                     (1 files differ)
mww3_test_04/./work_PR2_UQ_f                     (1 files differ)
mww3_test_04/./work_PR3_UNO_f                     (1 files differ)
mww3_test_04/./work_PR2_UQ_MPI_d                     (1 files differ)
mww3_test_04/./work_PR3_UQ_b                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_04/./work_PR2_UQ_MPI_b                     (1 files differ)
mww3_test_04/./work_PR2_UNO_MPI_d                     (1 files differ)
mww3_test_04/./work_PR2_UQ_g                     (1 files differ)
mww3_test_04/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_04/./work_PR1_b                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_d_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_d                     (1 files differ)
mww3_test_04/./work_PR1_g                     (1 files differ)
mww3_test_04/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_04/./work_PR3_UNO_g_c                     (1 files differ)
mww3_test_04/./work_PR1_MPI_e                     (1 files differ)
mww3_test_04/./work_PR3_UNO_a_c                     (2 files differ)
mww3_test_04/./work_PR3_UNO_MPI_b_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_a                     (2 files differ)
mww3_test_04/./work_PR2_UNO_b                     (1 files differ)
mww3_test_04/./work_PR3_UQ_MPI_NC_b_c                     (1 files differ)
mww3_test_04/./work_PR3_UNO_b_c                     (1 files differ)
mww3_test_04/./work_PR3_UNO_MPI_c                     (1 files differ)
mww3_test_04/./work_PR1_MPI_d                     (1 files differ)
mww3_test_04/./work_PR3_UNO_f_c                     (1 files differ)
mww3_test_04/./work_PR3_UQ_f                     (1 files differ)
mww3_test_04/./work_PR3_UNO_g                     (1 files differ)
mww3_test_06/./work_PR3_UNO_d                     (1 files differ)
mww3_test_06/./work_PR3_UQ_d                     (1 files differ)
mww3_test_07/./work_PR3_UQ                     (1 files differ)
mww3_test_09/./work_MPI                     (3 files differ)
ww3_ta1/./work_UPD5_U                     (2 files differ)
ww3_ta1/./work_UPD3_U                     (2 files differ)
ww3_ta1/./work_UPD0F_O                     (2 files differ)
ww3_ta1/./work_UPD2_U                     (2 files differ)
ww3_ta1/./work_UPD3_U_cap                     (2 files differ)
ww3_ta1/./work_UPD0F_U                     (2 files differ)
ww3_ta1/./work_UPD5_U_cap                     (2 files differ)
ww3_ta1/./work_UPD2_O                     (2 files differ)
ww3_ta1/./work_UPD6_U                     (2 files differ)
ww3_ta1/./work_UPD6_O                     (2 files differ)
ww3_ta1/./work_UPD6_U_cap                     (2 files differ)
ww3_ta1/./work_UPD2_U_cap                     (2 files differ)
ww3_ta1/./work_UPD5_O                     (2 files differ)
ww3_ta1/./work_UPD3_O                     (2 files differ)
ww3_tbt1.1/./work_BT8                     (4 files differ)
ww3_tbt1.1/./work_BT8_MPI                     (4 files differ)
ww3_tbt1.1/./work_BT9                     (4 files differ)
ww3_tbt1.1/./work_BT9_MPI                     (4 files differ)
ww3_tbt2.1/./work_BT8                     (3 files differ)
ww3_tbt2.1/./work_BT8_MPI                     (3 files differ)
ww3_tbt2.1/./work_BT9                     (3 files differ)
ww3_tbt2.1/./work_BT9_MPI                     (3 files differ)
ww3_tic1.1/./work_5km_IC1                     (3 files differ)
ww3_tic1.1/./work_1000m_nondisp_IC2_ifr                     (3 files differ)
ww3_tic1.1/./work_350m_IC3                     (3 files differ)
ww3_tic1.1/./work_IC3_CGICE_CHENG                     (3 files differ)
ww3_tic1.1/./work_IC5_M3                     (3 files differ)
ww3_tic1.1/./work_IC5_M2                     (3 files differ)
ww3_tic1.1/./work_IC4_M6                     (3 files differ)
ww3_tic1.1/./work_2500m_IC1                     (3 files differ)
ww3_tic1.1/./work_IC4_M2                     (3 files differ)
ww3_tic1.1/./work_IC4_M5                     (3 files differ)
ww3_tic1.1/./work_IC4_M9                     (3 files differ)
ww3_tic1.1/./work_IC4_M3                     (3 files differ)
ww3_tic1.1/./work_IC3_CGOW_NOCHENG_MPI                     (3 files differ)
ww3_tic1.1/./work_IC4_M10                     (3 files differ)
ww3_tic1.1/./work_IC4_M1                     (3 files differ)
ww3_tic1.1/./work_IC4_M8                     (3 files differ)
ww3_tic1.1/./work_IC3_CGOW_CHENG_MPI                     (3 files differ)
ww3_tic1.1/./work_1000m_IC2_nrl                     (3 files differ)
ww3_tic1.1/./work_1000m_IC3                     (3 files differ)
ww3_tic1.1/./work_100m_IC1                     (3 files differ)
ww3_tic1.1/./work_IC4_M6H                     (3 files differ)
ww3_tic1.1/./work_1000m_IC1                     (3 files differ)
ww3_tic1.1/./work_IC4_M7                     (3 files differ)
ww3_tic1.1/./work_IC5_M1                     (3 files differ)
ww3_tic1.1/./work_IC4_M4                     (3 files differ)
ww3_tic1.1/./work_1000m_nondisp_IC2_nrl                     (3 files differ)
ww3_tic1.1/./work_IC3_CGICE_CHENG_MPI                     (3 files differ)
ww3_tic1.1/./work_10km_IC1                     (3 files differ)
ww3_tic1.1/./work_350m_IC1                     (3 files differ)
ww3_tic1.1/./work_IC4_M6L                     (3 files differ)
ww3_tic1.1/./work_20km_IC1                     (3 files differ)
ww3_tic1.1/./work_2500m_IC3                     (3 files differ)
ww3_tic1.1/./work_IC4_M10_icenum                     (3 files differ)
ww3_tic1.2/./work_IC3_B2.5k                     (3 files differ)
ww3_tic1.2/./work_IC3_B2.5k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_A1.0k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_V1_h                     (3 files differ)
ww3_tic1.2/./work_IC3_A2.5k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_CHENG                     (3 files differ)
ww3_tic1.2/./work_IC3_B1.0k                     (3 files differ)
ww3_tic1.2/./work_IC3_B1.0k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_B0.5k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_A0.5k_MPI                     (3 files differ)
ww3_tic1.2/./work_IC3_A1.0k                     (3 files differ)
ww3_tic1.2/./work_IC3_A0.5k                     (3 files differ)
ww3_tic1.2/./work_IC3_A2.5k                     (3 files differ)
ww3_tic1.2/./work_IC3_V1_G                     (3 files differ)
ww3_tic1.2/./work_IC3_B0.5k                     (3 files differ)
ww3_tic1.3/./work_IC3_2.5k_PR2_UQ                     (3 files differ)
ww3_tic1.3/./work_IC3_V1_h                     (3 files differ)
ww3_tic1.3/./work_IC3_0.5k_PR3_UQ                     (3 files differ)
ww3_tic1.3/./work_IC3_2.5k_PR2_UNO                     (3 files differ)
ww3_tic1.3/./work_IC3_0.5k_PR2_UQ                     (3 files differ)
ww3_tic1.3/./work_IC3_0.5k_PR1                     (3 files differ)
ww3_tic1.3/./work_IC3_CHENG                     (3 files differ)
ww3_tic1.3/./work_IC3_2.5k_PR1                     (3 files differ)
ww3_tic1.3/./work_IC3_0.5k_PR2_UNO                     (3 files differ)
ww3_tic1.3/./work_IC3_0.5k_PR3_UNO                     (3 files differ)
ww3_tic1.3/./work_IC3_2.5k_PR3_UNO                     (3 files differ)
ww3_tic1.3/./work_IC3_2.5k_PR3_UQ                     (3 files differ)
ww3_tic1.3/./work_IC3_V1_G                     (3 files differ)
ww3_tic2.1/./work_IC1                     (3 files differ)
ww3_tic2.2/./work_IC2IS2                     (3 files differ)
ww3_tic2.3/./work_IC2IS2scat                     (3 files differ)
ww3_tic2.3/./work_IC2IS2dissip                     (3 files differ)
ww3_tic2.3/./work_IC2IS2creep                     (3 files differ)
ww3_tp1.1/./work_PR2_UQ                     (1 files differ)
ww3_tp1.1/./work_PR3_UNO                     (1 files differ)
ww3_tp1.1/./work_PR3_UQ                     (1 files differ)
ww3_tp1.1/./work_PR1                     (1 files differ)
ww3_tp1.1/./work_PR2_UNO                     (1 files differ)
ww3_tp1.3/./work_PR2_UQ                     (1 files differ)
ww3_tp1.3/./work_PR3_UNO                     (1 files differ)
ww3_tp1.3/./work_PR3_UQ                     (1 files differ)
ww3_tp1.3/./work_PR1                     (1 files differ)
ww3_tp1.3/./work_PR2_UNO                     (1 files differ)
ww3_tp1.4/./work_PR2_UQ                     (4 files differ)
ww3_tp1.4/./work_PR3_UNO                     (4 files differ)
ww3_tp1.4/./work_PR3_UQ                     (4 files differ)
ww3_tp1.4/./work_PR1                     (4 files differ)
ww3_tp1.4/./work_PR2_UNO                     (4 files differ)
ww3_tp1.5/./work_PR2_UQ                     (4 files differ)
ww3_tp1.5/./work_PR3_UNO                     (4 files differ)
ww3_tp1.5/./work_PR3_UQ                     (4 files differ)
ww3_tp1.5/./work_PR1                     (4 files differ)
ww3_tp1.5/./work_PR2_UNO                     (4 files differ)
ww3_tp1.6/./work_PR2_UQ                     (5 files differ)
ww3_tp1.6/./work_PR3_UNO                     (5 files differ)
ww3_tp1.6/./work_PR3_UQ                     (5 files differ)
ww3_tp1.6/./work_PR1                     (5 files differ)
ww3_tp1.6/./work_PR2_UNO                     (5 files differ)
ww3_tp1.7/./work_PR3_UQ                     (1 files differ)
ww3_tp1.8/./work_BJ_PR3_UQ                     (1 files differ)
ww3_tp1.8/./work_PR3_UQ                     (1 files differ)
ww3_tp1.9/./work_PR3_UQ                     (1 files differ)
ww3_tp1.9/./work_PR3_UQ_MPI                     (1 files differ)
ww3_tp2.10/./work_MPI                     (1 files differ)
ww3_tp2.11/./work_MPI                     (1 files differ)
ww3_tp2.17/./work_b                     (2 files differ)
ww3_tp2.17/./work_ma1                     (1 files differ)
ww3_tp2.17/./work_c                     (2 files differ)
ww3_tp2.17/./work_mc                     (1 files differ)
ww3_tp2.17/./work_ma                     (1 files differ)
ww3_tp2.17/./work_mc1                     (1 files differ)
ww3_tp2.17/./work_mb                     (1 files differ)
ww3_tp2.17/./work_a                     (2 files differ)
ww3_tp2.2/./work_PR2_UQ                     (4 files differ)
ww3_tp2.2/./work_PR3_UNO                     (4 files differ)
ww3_tp2.2/./work_PR2_UQ_MPI                     (4 files differ)
ww3_tp2.2/./work_PR1_MPI_BIN2NC                     (4 files differ)
ww3_tp2.2/./work_PR3_UNO_MPI                     (4 files differ)
ww3_tp2.2/./work_PR3_UQ                     (4 files differ)
ww3_tp2.2/./work_PR3_UQ_MPI                     (4 files differ)
ww3_tp2.2/./work_PR2_UNO_MPI                     (4 files differ)
ww3_tp2.2/./work_PR1                     (4 files differ)
ww3_tp2.2/./work_PR2_UNO                     (4 files differ)
ww3_tp2.2/./work_PR1_MPI                     (4 files differ)
ww3_tp2.21/./work_ma                     (1 files differ)
ww3_tp2.21/./work_mb                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (1 files differ)
ww3_tp2.6/./work_ST0                     (1 files differ)
ww3_tp2.6/./work_ST4                     (1 files differ)
ww3_tp2.7/./work_ST0                     (1 files differ)
ww3_tp2.8/./work_PR3_UQ                     (1 files differ)
ww3_tr1/./work_std                     (1 files differ)
ww3_tr1/./work                     (1 files differ)
ww3_tr1/./work_bndin                     (1 files differ)
ww3_ts1/./work_ST4_TSA                     (2 files differ)
ww3_ts1/./work_ST4_T500                     (2 files differ)
ww3_ts1/./work_ST1_RWND                     (2 files differ)
ww3_ts1/./work_ST4_WRT                     (2 files differ)
ww3_ts1/./work_ST4_T700                     (2 files differ)
ww3_ts1/./work_ST6                     (2 files differ)
ww3_ts1/./work_NL5                     (1 files differ)
ww3_ts1/./work_ST4_GMD                     (2 files differ)
ww3_ts1/./work_ST1                     (2 files differ)
ww3_ts1/./work_ST4                     (2 files differ)
ww3_ts1/./work_ST3                     (2 files differ)
ww3_ts1/./work_ST2                     (2 files differ)
ww3_ts2/./work_ST6_PR2_UQ                     (3 files differ)
ww3_ts2/./work_ST6_PR3_UNO                     (3 files differ)
ww3_ts2/./work_ST2_PR2_UNO                     (3 files differ)
ww3_ts2/./work_ST6_PR2_UNO                     (3 files differ)
ww3_ts2/./work_ST3_PR3_UQ                     (3 files differ)
ww3_ts2/./work_ST1_PR3_UQ                     (3 files differ)
ww3_ts2/./work_ST2_PR3_UNO                     (3 files differ)
ww3_ts2/./work_ST1_PR3_UNO                     (3 files differ)
ww3_ts2/./work_ST2_PR3_UQ                     (3 files differ)
ww3_ts2/./work_ST3_PR2_UNO                     (3 files differ)
ww3_ts2/./work_ST4_PR3_UNO                     (3 files differ)
ww3_ts2/./work_ST1_PR1                     (3 files differ)
ww3_ts2/./work_ST3_PR2_UQ                     (3 files differ)
ww3_ts2/./work_ST3_PR3_UNO                     (3 files differ)
ww3_ts2/./work_ST4_PR1                     (3 files differ)
ww3_ts2/./work_ST4_PR2_UQ                     (3 files differ)
ww3_ts2/./work_ST6_PR1                     (3 files differ)
ww3_ts2/./work_ST6_PR3_UQ                     (3 files differ)
ww3_ts2/./work_ST2_PR2_UQ                     (3 files differ)
ww3_ts2/./work_ST4_PR2_UNO                     (3 files differ)
ww3_ts2/./work_ST1_PR2_UNO                     (3 files differ)
ww3_ts2/./work_ST2_PR1                     (3 files differ)
ww3_ts2/./work_ST1_PR2_UQ                     (3 files differ)
ww3_ts2/./work_ST3_PR1                     (3 files differ)
ww3_ts2/./work_ST4_PR3_UQ                     (3 files differ)
ww3_ts3/./work_ST4_PR3_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR1_OMP                     (1 files differ)
ww3_ts3/./work_ST1_PR1_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST4_FLD2_MPI                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR1_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR1_OMP                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST4_PR3_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST4_PR3_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST4_PR3_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST4_FLD1_MPI                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST1_PR1_MPI                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR1_MPI                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST4_PR3_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST2_PR1_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR1_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UQ_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR1_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST3_PR1_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR1_MPI                     (1 files differ)
ww3_ts3/./work_ST3_PR3_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR1_OMP                     (1 files differ)
ww3_ts3/./work_ST4_PR2_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST4_PR3_UNO_MPI                     (1 files differ)
ww3_ts3/./work_ST1_PR3_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST6_PR2_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST2_PR3_UNO_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST1_PR1_OMP                     (1 files differ)
ww3_ts3/./work_ST2_PR1_MPI                     (1 files differ)
ww3_ts3/./work_ST3_PR2_UQ_MPI_OMPH                     (1 files differ)
ww3_ts3/./work_ST1_PR2_UNO_OMP                     (1 files differ)
ww3_ts3/./work_ST3_PR1_MPI                     (1 files differ)
ww3_ts3/./work_ST6_PR3_UQ_MPI                     (1 files differ)
ww3_ts3/./work_ST2_PR2_UNO_MPI                     (1 files differ)
ww3_ufs1.1/./work_d                     (1 files differ)
ww3_ufs1.1/./work_c_nth                     (1 files differ)
ww3_ufs1.1/./work_c                     (1 files differ)
ww3_ufs1.1/./work_unstr_c                     (3 files differ)
ww3_ufs1.1/./work_unstr_a                     (3 files differ)
ww3_ufs1.1/./work_unstr_b                     (3 files differ)
ww3_ufs1.1/./work_c_npl                     (1 files differ)
ww3_ufs1.2/./work_b                     (5 files differ)
ww3_ufs1.2/./work_c                     (4 files differ)
ww3_ufs1.2/./work_l                     (1 files differ)
ww3_ufs1.2/./work_a                     (5 files differ)
```
[matrixCompFull.txt](https://github.com/user-attachments/files/22728194/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/user-attachments/files/22728195/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/user-attachments/files/22728196/matrixDiff.txt)

Commit:
Add explicit EXTERNAL declarations to resolve Intel compiler warning #8889 for WW3 prepost build

Commit:
Add explicit EXTERNAL declarations to resolve Intel compiler warning #8889 for ufs_gfs debug build

Commit:
fix wave postprocess errors and warnings for global workflow debug mode

JessicaMeixner-NOAA commented:
@mingchen-NOAA  - Can you please explain which updates fix the wavepost issues as there's quite a bit going on here?  

In the future, please consider putting updates like this for multiple things in multiple PRs.

JessicaMeixner-NOAA commented:
Regtests are starting on ursa, hopefully we can get them through before machine goes offline-  i'll also test on Hercules as a back-up in case.

mingchen-NOAA commented:
> @mingchen-NOAA - Can you please explain which updates fix the wavepost issues as there's quite a bit going on here?
> 
> In the future, please consider putting updates like this for multiple things in multiple PRs.

@JessicaMeixner-NOAA I’ve updated the PR description to clarify which updates address which issue and added links to the relevant commits for clarity. I hope this helps.

In the future, I’ll separate multiple fixes into individual PRs to keep each one simpler and easier to review. Thank you for the helpful advice.

JessicaMeixner-NOAA commented:
Thanks for adding the extra descriptions @mingchen-NOAA ! I'll post the results of the regrestests as soon as I have them.

JessicaMeixner-NOAA commented:
[matrixCompFull.txt](https://github.com/user-attachments/files/22752717/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/user-attachments/files/22752718/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/user-attachments/files/22752719/matrixDiff.txt)

JessicaMeixner-NOAA reviewed (approved):
Thanks @mingchen-NOAA  for helping with additional clean-up and the debug failures.
